src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -54,88 +57,175 @@ private boolean compliesToMidsTwo(DigitalSpecimen attributes) {
+            && georeference.getDwcCoordinateUncertaintyInMeters() != null
+            && georeference.getDwcCoordinatePrecision() != null)
+            || (isValid(georeference.getDwcFootprintWkt())
+            && isValid(georeference.getDwcFootprintSrs()));
+      }
+    }
+    return false;
+  }
+
+  private boolean taxValuesComply(DigitalSpecimen attributes) {
+    var hasScientificNameId = false;
+    var hasIdentifiedById = false;
+    for (var identifications : attributes.getDwcIdentification()) {
+      if (isValid(identifications.getDwcIdentificationID())) {
+        hasIdentifiedById = true;
+      }
+      for (var taxonIdentification : identifications.getTaxonIdentifications()) {
+        if (isValid(taxonIdentification.getDwcScientificNameId())) {
+          hasScientificNameId = true;
+        }
+      }
+    }
+    return hasScientificNameId && hasIdentifiedById;
+  }
+
   private boolean compliesToMidsTwoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && (attributes.getOdsHasMedia() != null && attributes.getOdsHasMedia())
         && isQualitativeLocationValid(attributes)
         && isQuantitativeLocationValid(attributes)
-        && (occurrenceIsPresent(attributes) && isValid(
-        (attributes.getOccurrences().get(0).getDwcEventDate())))
-        && (occurrenceIsPresent(attributes) && isValid(
-        attributes.getOccurrences().get(0).getDwcFieldNumber()))
-        && isValid(attributes.getDwcRecordedBy());
+        && (occurrenceIsPresent(attributes) && hasValue(
+        attributes.getDwcOccurrence().get(0).getDwcEventDate(),
+        attributes.getDwcOccurrence().get(0).getDwcVerbatimEventDate(),
+        convertInteger(attributes.getDwcOccurrence().get(0).getDwcYear())))
+        && (occurrenceIsPresent(attributes) &&
+        hasValue(attributes.getDwcOccurrence().get(0).getDwcFieldNumber(),
+            attributes.getDwcOccurrence().get(0).getDwcRecordNumber()))
+        && hasValue(attributes.getDwcRecordedBy(), attributes.getDwcRecordedById());
+  }
+
+  private String convertInteger(Integer integer) {
+    if (integer != null) {
+      return integer.toString();
+    } else {
+      return null;
+    }
+  }
+
+  private boolean hasValue(String... terms) {
+    for (var term : terms) {
+      if (isValid(term)) {
+        return true;
+      }
+    }
+    return false;
   }
 
 
   private boolean compliesToMidsTwoPaleoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && isStratigraphyValid(attributes)
         && isQualitativeLocationValid(attributes)
         && isQuantitativeLocationValid(attributes);
   }
 
   private boolean isQuantitativeLocationValid(DigitalSpecimen digitalSpecimen) {
```

Comment:
change to quantitativeLocationIsValid? Be direct!

Reply:
Changed them